src/main/java/build/buildfarm/instance/shard/DistributedStateCreator.java

```diff
@@ -143,6 +146,18 @@ static List<String> getQueueHashes(RedisClient client, String queueName) throws
                 jedis, RedisHashtags.existingHash(queueName)));
   }
 
+  private static int getMaxNodeQueueDepth(RedisClient client, int maxQueueDepth)
+      throws IOException {
+    return client.call(
+        jedis -> {
+          if (jedis instanceof JedisCluster) {
+            JedisCluster cluster = (JedisCluster) jedis;
+            return maxQueueDepth / cluster.getClusterNodes().keySet().size();
+          }
+          return 1;
```

Comment:
This is incorrect - it should be the maxQueueDepth, not 1. Can you check to see if there's coverage in here?